libqpdf/QPDFJob.cc

```diff
@@ -2350,30 +2350,33 @@ QPDFJob::handleTransformations(QPDF& pdf)
         for (auto& ph: dh.getAllPages()) {
             ++pageno;
             QPDFObjectHandle page = ph.getObjectHandle();
-            for (auto& iter2: ph.getImages()) {
-                std::string name = iter2.first;
-                QPDFObjectHandle& image = iter2.second;
-                ImageOptimizer* io = new ImageOptimizer(
-                    *this,
-                    m->oi_min_width,
-                    m->oi_min_height,
-                    m->oi_min_area,
-                    image);
-                std::shared_ptr<QPDFObjectHandle::StreamDataProvider> sdp(io);
-                if (io->evaluate(
-                        "image " + name + " on page " +
-                        std::to_string(pageno))) {
-                    QPDFObjectHandle new_image = pdf.newStream();
-                    new_image.replaceDict(image.getDict().shallowCopy());
-                    new_image.replaceStreamData(
-                        sdp,
-                        QPDFObjectHandle::newName("/DCTDecode"),
-                        QPDFObjectHandle::newNull());
-                    ph.getAttribute("/Resources", true)
-                        .getKey("/XObject")
-                        .replaceKey(name, new_image);
-                }
-            }
+            ph.forEachImage(
+                true,
+                [this, pageno, &pdf](
+                    QPDFObjectHandle& obj,
+                    QPDFObjectHandle& xobj_dict,
+                    std::string const& key) {
+                    QPDFObjectHandle& image = obj;
```

Comment:
Sorry, also no need to create a new to create a new ref to obj.

(As an aside, `(auto& image, auto& xobj_dict, auto const& key)` also works.)